galsim/compound.py

```diff
@@ -902,3 +905,263 @@ def __setstate__(self, d):
 _galsim.SBFourierSqrt.__setstate__ = lambda self, state: 1
 _galsim.SBFourierSqrt.__repr__ = lambda self: \
         'galsim._galsim.SBFourierSqrt(%r, %r)'%self.__getinitargs__()
+
+
+class RandomWalk(Sum):
+    """
+
+    A class for generating a set of point sources distributed using a random
+    walk.  Uses of this profile include representing an "irregular" galaxy, or
+    adding this profile to an Exponential to represent knots of star formation.
+
+    Random walk profiles have "shape noise" that depends on the number of point
+    sources used.  For example, with 100 points the shape noise is g~0.05, and
+    this will decrease as more points are added.  The profile can be sheared to
+    give additional ellipticity, for example to follow that of an associated
+    disk.
+
+    We use the analytic approximation of an infinite number of steps, which is
+    a good approximation even if the desired number of steps were less than 10.
+
+    The requested half light radius (hlr) should be thought of as a rough
+    value.  With a finite number point sources the actual realized hlr will be
+    noisy.
+
+    Initialization
+    --------------
+    @param  npoints                 Number of point sources to generate.
+    @param  half_light_radius       Half light radius of the distribution of
+                                    points.  This is the mean half light
+                                    radius produced by an infinite number of
+                                    points.  A single instance will be noisy.
+    @param  flux                    Optional total flux in all point sources.
+                                    [default: 1]
+    @param  rng                     Optional random number generator. Can be
+                                    any galsim.BaseDeviate.  If not sent or
+                                    None, the rng is created internally.
+                                    [default: None]
+    @param  gsparams                Optional GSParams for the gaussians
+                                    representing each point source.
+                                    [default: None]
+
+    Methods
+    -------
+
+    This class inherits from galsim.Sum. Additional methods are
+    
+        calculateHLR:
+            Calculate the actual half light radius of the generated points
+
+    There are also "getters",  implemented as read-only properties
+
+        .npoints
+        .input_half_light_radius
+        .flux
+        .gaussians
+            The list of galsim.Gaussian objects representing the points
+        .points
+            The array of x,y offsets used to create the point sources
+
+    Notes
+    -----
+
+    - The algorithm is a modified version of that presented in
+
+          https://arxiv.org/abs/1312.5514v3.
```

Comment:
Super minor point, but I recommend omitting the . at the end of the URL (if someone just copies this whole line and doesn't notice it, they'll get a "can't find this preprint" blah blah error).